pkg/querier/queryrange/volume.go

```diff
@@ -212,3 +215,75 @@ func toPrometheusData(series map[string][]logproto.LegacySample, aggregateBySeri
 		Result:     result,
 	}
 }
+
+type aggregatedMetricQuery struct {
+	matchers    []*labels.Matcher
+	aggregateBy string
+	start       time.Time
+	end         time.Time
+}
+
+func (a *aggregatedMetricQuery) buildBaseQueryString(
+	idxKey string,
+	idxMatch string,
+	idxVal string,
+	serviceNamePresent bool,
+) string {
+	aggBy := a.aggregateBy
+	if aggBy == "" {
+		if serviceNamePresent {
+			aggBy = push.LabelServiceName
+		} else {
+			aggBy = idxKey
+		}
+	}
+
+	if serviceNamePresent {
+		return fmt.Sprintf(
+			`sum by (%s) (sum_over_time({%s%s"%s"} | logfmt`,
+			aggBy,
+			idxKey,
+			idxMatch,
+			idxVal,
+		)
+	}
+
+	return fmt.Sprintf(
+		`sum by (%s) (sum_over_time({%s=~".+"} | logfmt`,
+		aggBy,
+		push.AggregatedMetricLabel,
+	)
+}
+
+func (a *aggregatedMetricQuery) BuildQuery() string {
+	var idxKey, idxVal, idxMatch string
+	filters := []string{}
+	serviceNamePresent := false
+
+	for i, matcher := range a.matchers {
```

Comment:
Is len(matchers) ever 0? If not, it might be good to pull the `if i == 0` clause out for clarity